components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -331,19 +334,28 @@ open class GleanInternalAPI internal constructor () {
             return false
         }
 
-        var hasPingContent = false
+        val pingSerializationTasks: MutableList<Job> = mutableListOf()
```

Comment:
It just occurred to me that this is not enough to address the case of multiple consecutive calls to `Glean.sendPings`. Think of this case:

-  `Glean.sendPings` is called once (slower)
- some ping is collected, an upload task is not enqueued yet
-  `Glean.sendPings` is called a second time (faster)
- the first upload task is still not enqueued, the second one is enqueued
- the first upload task is finally enqueued, kills the second one and starts

I think this is out of scope for this PR, but maybe this could trigger duplicates or spurious ping sent. We should probably *enqueue* upload tasks, not replace the current ones

Reply:
So make Glean.sendPings `@Synchronized` ?   We don't need to enqueue multiple upload tasks, any single one would upload all serialized pings so there is no point in having more than one enqueued.

Reply:
Mh, maybe this is worth filing a bug. Let's not tackle this in the current PR: I'm not sure `synchronized` is the best option here, though it could solve this problem.

Reply:
I'll file an investigation bug and maybe we can set up a test to try and incite this issue to see if it actually occurs or not.

Reply:
Bug filed [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1541022)